### DISCUSSION TITLE
What version for Windows XP?

### DISCUSSION THREAD

**Discussion by mleese** [ImageMagick/ImageMagick#2376] [Help] [answered]
Hi,

I need to install ImageMagick under Windows XP.  What is the most recent version that runs under Windows XP?

From https://legacy.imagemagick.org/script/download.php I can download version 6.9.11-25, but the commentary on the page suggests that I need something older.

Many thanks for any help you can give.

Regards,
Martin

**Comment by mleese** [comment]
Attempting to install or use version 6.9.11-25 (or version 7.0.10-24) results in the error "... is not a valid Win32 application".  So definitely need something older.

**Reply by fmw42** [reply]
Which binary did you try? It looks like your OS is Win32. Is that correct?  If so, then use one of the Win32 binaries on that page. Or did you do that and it does not work? Also see the note about Visual C++ further down the page.

**Comment by dlemstra** [comment]
Support for Windows XP has been stopped early this year. Windows stopped supporting it 6 years ago.

**Comment by mleese** [comment]
Hi fmw42,

Thanks for trying to help.  I tried the following four binaries and two zip files:  
    ImageMagick-6.9.11-25-Q8-x86-dll.exe
    ImageMagick-6.9.11-25-Q8-x86-static.exe
    ImageMagick-6.9.11-25-portable-Q8-x86.zip

    ImageMagick-7.0.10-25-Q8-x86-dll.exe
    ImageMagick-7.0.10-25-Q8-x86-static.exe
    ImageMagick-7.0.10-25-portable-Q8-x86.zip

Yes, my OS is Win32.  All four binaries gave the error, "... is not a valid Win32 application".  The zip files unzipped OK, but gave the same error when I tried to run convert.exe or magick.exe (as appropriate).

With regard to the note about Visual C++ further down the page,  I first searched my C: drive for "vcomp120.dll" without success.  I then downloaded and installed vc_redist.x86.exe from the page:  
    https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

It installed OK, but a second search for "vcomp120.dll" was unsuccessful and running ImageMagick-6.9.11-25-Q8-x86-dll.exe still failed with the same error.

I feel I should point out that the commentary on the page:  
    https://legacy.imagemagick.org/script/download.php

lists the OSs that version 6.9.11-25 runs under.  Windows XP is *not* listed.  I am puzzled why you think it should.

**Reply by fmw42** [reply]
dlemstra, one of the developers said: "Support for Windows XP has been stopped early this year. Windows stopped supporting it 6 years ago.".  So I suspect ImageMagick has pulled binary installs for XP or at least none are compatible. Perhaps use an older binary. See https://ftp.icm.edu.pl/packages/ImageMagick/binaries/

**Comment by mleese** [comment]
The most recent version of ImageMagick that runs under Windows XP is 6.9.11-19.  This version is dated June 12, 2020.

**Comment by ivanooi** [comment]
tried 7.0.9. no issues for WinXP

**Reply by duansl** [reply]
hi,ivanooi
 Could you send me a package that can be used by win XP,
my email   duanshaolong@gmail.com
thanks

**Comment by mleese** [comment]
Hi ivanoo!,

To settle on version 6.9.11-19 as the latest version of ImageMagick that runs under Windows XP, I downloaded .exe files from:  
    https://ftp.icm.edu.pl/packages/ImageMagick/binaries/

Version ImageMagick-6.9.11-19-Q8-x86-dll.exe installed, version ImageMagick-6.9.11-20-Q8-x86-dll.exe did not install.  Note that I was running the .exe installation files, not ImageMagick per se.  Perhaps that explains the difference in our experiences.

Regards,
Martin

**Reply by duansl** [reply]
hi mleese,
Could you send me a package that can be used by win XP
my email duanshaolong@gmail.com
thanks

**Comment by ivanooi** [comment]
i see...I preferred the portable version. I think that the difference.. I had checked with the latest build 7.0.10-19 which is compatible with XP. build 20 will not compatible with XP anymore. so, 7.0.10 build 19 will be the last build for WinXP.

**Comment by mleese** [comment, marked as answer]
Hi ivanooi,

I can confirm that ImageMagick-7.0.10-19-portable-Q16-x86.zip runs and ImageMagick-7.0.10-20-portable-Q16-x86.zip does not run under Windows XP.  

I actually prefer a Q8 installation; I have used Q16 in the past and got into a mess.  So I downloaded ImageMagick-6.9.11-29-portable-Q8-x86.zip, this being the latest version I could find that was Q8.  However it did not run, giving the error:  
    convert.exe is not a valid Win32 application.

I find this puzzling, but no matter.  I have a Version 6 Q8 and a Version 7 Q16 that run under Windows XP.  This is sufficient.

Many thanks for your help.

Regards,
Martin

**Comment by mleese** [comment]
Hi S,

The binaries are still at:  
    https://ftp.icm.edu.pl/packages/ImageMagick/binaries/

I had more success accessing the website using an FTP client (as opposed to a Web browser).

I have the following two packages:  
    ImageMagick-6.9.11-19-Q8-x86-dll.exe  (30 MBytes)
    ImageMagick-7.0.10-19-portable-Q16-x86.zip (79 MBytes)

Regards,
Martin

**Reply by sergeevabc** [reply]
> The binaries are still at:
> https://ftp.icm.edu.pl/packages/ImageMagick/binaries/

Is that so? I see neither 6.9.11.* nor 7.0.10.* ([screenshot][1]).
Github allows to upload ZIP, be kind to attach the latter somewhere here.

[1]: https://user-images.githubusercontent.com/373718/120557076-eb299080-c3ec-11eb-867a-6f71b4c6e770.png

**Reply by sergeevabc** [reply]
@mleese, just drag zip into the comment box and it will be uploaded right away.

**Reply by duansl** [reply]
hi mleese,
Could you send me a package that can be used by win XP
my email duanshaolong@gmail.com
thanks

**Comment by mleese** [comment]
Hi S,

If nobody else does this earlier, I can do this after my local library re-opens.  (They have high-speed Internet.)  This is scheduled to happen soon after June 10, 2021.

Regards,
Martin

**Reply by mleese** [reply]
Hi S,

Here is:  
    ImageMagick-6.9.11-19-Q8-x86-dll.exe.zip


    ImageMagick-7.0.10-19-portable-Q16-x86.zip


(although I cannot see them).  If this hasn't worked then we can try again.

Regards,
Martin

**Reply by DevanWolf** [reply]
Please upload the files!

**Reply by mleese** [reply]
When I try to upload the files it says "is not included in the list".  Don't know why.

Sorry,
Martin

**Reply by DevanWolf** [reply]
> When I try to upload the files it says "is not included in the list". Don't know why.
> 
> Sorry,
> Martin

Try using MEGA, Google Drive, or Discord instead to upload.

**Reply by mleese** [reply]
I used DropBox:  

https://www.dropbox.com/s/zmrs5bv8uykvdhl/ImageMagick-6.9.11-19-Q8-x86-dll.exe
https://www.dropbox.com/s/fah5x22k95t91n5/ImageMagick-7.0.10-19-portable-Q16-x86.zip

Regards,
Martin

**Reply by Thyristor-Chopper** [reply]
link doesn't work anymore, is there any reupload available?

**Comment by mleese** [comment]
Try:  

http://martinleese.epizy.com/MyTemporaryDownloads/#WINDOWSXP

Note you will need to download multiple files and stitch them back together using 7zip.

Regards,
Martin

**Reply by Thyristor-Chopper** [reply]
Oh, thanks.
